components/payments/internal/app/connectors/mangopay/client/client.go

```diff
@@ -0,0 +1,113 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/formancehq/stack/libs/go-libs/logging"
+	"github.com/zitadel/oidc/pkg/oidc"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+)
+
+type Client struct {
+	httpClient *http.Client
+
+	clientID string
+	apiKey   string
+	endpoint string
+
+	logger logging.Logger
+
+	accessToken          string
+	accessTokenExpiresAt time.Time
+}
+
+func newHTTPClient() *http.Client {
+	return &http.Client{
+		Timeout:   10 * time.Second,
+		Transport: otelhttp.NewTransport(http.DefaultTransport),
+	}
+}
+
+func NewClient(clientID, apiKey, endpoint string, logger logging.Logger) (*Client, error) {
+	endpoint = strings.TrimSuffix(endpoint, "/")
+
+	c := &Client{
+		httpClient: newHTTPClient(),
+
+		clientID: clientID,
+		apiKey:   apiKey,
+		endpoint: endpoint,
+
+		logger: logger,
+	}
+
+	if err := c.login(context.Background()); err != nil {
+		return nil, err
+	}
+
+	return c, nil
+}
+
+func (c *Client) login(ctx context.Context) error {
```

Comment:
You can use : https://pkg.go.dev/golang.org/x/oauth2/clientcredentials

Reply:
Done thanks !